Testnet/StandardToken/StandardToken/StandardToken.cs

```diff
@@ -163,6 +163,197 @@ public struct ApprovalLog
+/// </summary>
+public interface IBurnableWithMetadata
+{
+    /// <summary>
+    /// A user can burn tokens if the have the requisite balance available.
+    /// Burnt tokens are permanently removed from the total supply and cannot be retrieved.
+    /// </summary>
+    /// <remarks>Emits a TransferLog event with the 'to' address set to the zero address.
+    /// Additionally emits a BurnMetadata event containing the metadata string.</remarks>
+    /// <param name="amount">The quantity of tokens to be burnt.</param>
+    /// <param name="metadata">Additional data to be recorded with the burn.
+    /// The structure and interpretation of this data is unspecified here.</param>
+    bool BurnWithMetadata(UInt256 amount, string metadata);
+}
+
+[Deploy]
+public class StandardTokenEnhanced : StandardToken, IOwnable, IMintable, IBurnable, IBurnableWithMetadata
+{
+    public StandardTokenEnhanced(ISmartContractState smartContractState, UInt256 totalSupply, string name, string symbol, byte decimals) : base(smartContractState, totalSupply, name, symbol, decimals)
+    {
+        this.Owner = Message.Sender;
+    }
+
+    /// <inheritdoc />
+    public Address Owner
+    {
+        get => State.GetAddress(nameof(this.Owner));
+        private set => State.SetAddress(nameof(this.Owner), value);
+    }
+
+    /// <inheritdoc />
+    public void OnlyOwner()
+    {
+        Assert(IsOwner(), "Only the owner can call this method");
+    }
+
+    /// <inheritdoc />
+    public bool IsOwner()
+    {
+        return Message.Sender == Owner;
+    }
+
+    /// <inheritdoc />
+    public void RenounceOwnership()
+    {
+        OnlyOwner();
+
+        Address previousOwner = this.Owner;
+
+        this.Owner = Address.Zero;
+
+        Log(new OwnershipTransferred { PreviousOwner = previousOwner, NewOwner = Address.Zero });
+    }
+
+    /// <inheritdoc />
+    public void TransferOwnership(Address newOwner)
+    {
+        OnlyOwner();
+
+        Address previousOwner = this.Owner;
+
+        this.Owner = newOwner;
+
+        Log(new OwnershipTransferred { PreviousOwner = previousOwner, NewOwner = newOwner });
+    }
+
+    /// <inheritdoc />
+    public void Mint(Address account, UInt256 amount)
+    {
+        OnlyOwner();
+
+        UInt256 senderBalance = GetBalance(Message.Sender);
+
+        SetBalance(Message.Sender, senderBalance + amount);
+
+        Log(new TransferLog() { From = Address.Zero, To = account, Amount = amount });
+
+        this.TotalSupply += amount;
+    }
+
+    /// <inheritdoc />
+    public bool Burn(UInt256 amount)
+    {
+        if (TransferTo(Address.Zero, amount))
+        {
+            this.TotalSupply -= amount;
+
+            return true;
+        }
+
+        return false;
+    }
+
+    /// <inheritdoc />
+    public bool BurnWithMetadata(UInt256 amount, string metadata)
+    {
+        if (Burn(amount))
+        {
+            Log(new BurnMetadata() { From = Message.Sender, Amount = amount, Metadata = metadata });
```

Comment:
Up to you, not sure how ERC does it but the `From` and `Amount` fields will be logged within a `TransferLog` during the `Burn` => during the internal `TransferTo`.

So this log could only include the metadata, but 🤷‍♂️

Reply:
My rationale is that, if you were only searching for these metadata logs, you'd have sufficient information in them to do something actionable. And we need the amount as well as the metadata for interflux. So the duplication isn't optimal but it does simplify things elsewhere.

So it's mostly a choice of copy-pasting the transfer logic to do this specific type of burn, or have the two separate log entries with nearly identical information. I suppose having both logs does keep better compatibility with ERC20 though.

Reply:
Thats fine too, was more of just a callout. 

I am in favor of an added `SupplyLog` though or something named similarly, so TotalSupply changes are logged for watching. Opdex could use this for sure to track token supplies as they change rather than manual check after a Mint/Burn that has `Address.Zero` logged.

Reply:
I have added a log for tracking supply changes